### ISSUE TITLE
Custom utility classes - hover: screen: purged by Juice Inliner

### ISSUE THREAD

**Issue by landoDanziger** [maizzle/framework#722]
- Maizzle Version: v4.0.2
- CLI v1.4.0
- Node.js Version: v16.13.2

When adding custom utilities - responsive states are purged.

Example:
Copy button classes from promotional starter and assign to btn class in utilities:

.btn {
    @apply inline-block py-4 px-6 rounded text-base font-semibold text-center [text-decoration:none] text-white bg-indigo-800 hover:bg-indigo-700;
  } 

In dev localhost hover works, in maizzle build production hover: is removed. Initially discovered when trying to assign a custom font using screen: in custom utility class.

Hover states still function if styling is kept out of utility class. Added second button to promotional starter to test.

https://github.com/landoDanziger/utilityClassDemo

When inlineCss: false, is set - hover etc still works.

**Comment by cossssmin** [comment]
Thanks for providing the repo, this happens because of how we remove classes that have been inlined in Maizzle 4.0 (there was a change so that we can better support client targeting hacks).

So if you have something like this in your CSS files:

```postcss
@layer components {
  .btn {
    @apply bg-indigo-800 hover:bg-indigo-700;
}
```

... you'll need to set `removeStyleTags: true`:

```diff
module.exports = {
  build: {
    templates: {
      destination: {
        path: 'build_production',
      },
    },
  },
-  inlineCSS: true,
+  inlineCSS: {
+   removeStyleTags: true,
+  },
  removeUnusedCSS: true,
}
```

**Comment by landoDanziger** [comment]
Thanks Cosmin

Ah - this is related to that
https://github.com/maizzle/framework/discussions/563

At a first go - it works for the screen:font - would it work for hover: and active:

This comes back to the lack of a whitelist in Juice right?

Maybe for the buttons I will just bake it all into the button component using conditional logic - and take the screen:font-ptSerif for the win.

**Comment by landoDanziger** [comment]
What is peculiar about it is, it works when the styles are there regularly in the template. 
I was wondering if there was a way to import the utility shortcut first, then run the regular build.

Like an extra step or wait?

**Comment by landoDanziger** [comment]
I have been misunderstanding how Tailwinds Utilities function.

I imagined that they were shortcuts to avoid repeatedly writing out the same classes - which for web is more true - but they are really creating a class of tailwinds classes - so they won't transfer to the inliner in the same way as they are when written into the template. Got it. 

I think for the BTN classes - i'll be better to set them conditionally in the BTN component - for the screen:font-ptSerif class - I can see if I understand the tradeoff with the targeting hacks.

I did see a postCss plugin that unwraps nested rules - but I couldn't fully understand if that would make the styles more ingestible
https://github.com/postcss/postcss-nested

Anyway - guess this can be closed - I problem in my thinking.
